### PR TITLE
opentelemetry: added cmd flag to sanitize metric and label names

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
opentelemetry metric names and labels can not allowed symbols, which complicates using these metrics in Grafana and VMUI. Added `--opentelemetry.sanitizeMetrics` (which is set to false by default) to convert all now allowed symbols to `_` symbol.
Ported otel collector normalization rules
Partial fix for #6037

### CODE REVIEW

File: lib/protoparser/opentelemetry/stream/streamparser.go [hunk 1]
```diff
@@ -13,11 +14,17 @@ import (
...
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/prompbmarshal"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promrelabel"
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/common"
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/opentelemetry/pb"
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/writeconcurrencylimiter"
 )
 
+var (
+	// sanitizeMetricName controls sanitizing metric names ingested via OpenTelemetry protocol.
+	sanitizeMetricName = flag.Bool("opentelemetry.sanitizeMetricName", false, "Sanitize metric names for the ingested OpenTelemetry data")
```
Comment (f41gh7) [inline lib/protoparser/opentelemetry/stream/streamparser.go:25]: It'd great  to sanitize all label names as well. 

Without it, it ll be not possible to use such metrics with grafana prometheus datasource.

Comment (AndrewChubatiuk) [reply lib/protoparser/opentelemetry/stream/streamparser.go:25]: done

File: lib/protoparser/opentelemetry/stream/streamparser.go [hunk 2]
```diff
@@ -207,9 +220,15 @@ func (wr *writeContext) appendSampleWithExtraLabel(metricName, labelName, labelV
 
 // appendAttributesToPromLabels appends attributes to dst and returns the result.
 func appendAttributesToPromLabels(dst []prompbmarshal.Label, attributes []*pb.KeyValue) []prompbmarshal.Label {
+	var sanitizeFn func(string) string
```
Comment (f41gh7) [inline lib/protoparser/opentelemetry/stream/streamparser.go:223]: I think, it's better to specify default realization. It reduces complexity and better from reader PoV. 

```go
sanitizeFn := func(labelName string) string { return labelName }
```

 And override it if needed:
```go
if *sanitizeMetrics {
		sanitizeFn = promrelabel.SanitizeLabelName
}
```

**Review by f41gh7 — APPROVED** [review]
LGTM

File: lib/protoparser/opentelemetry/stream/streamparser.go [hunk 3]
```diff
@@ -290,6 +355,74 @@ func (wr *writeContext) parseRequestToTss(req *pb.ExportMetricsServiceRequest) {
...
+			return true
+		}
+	}
+	return false
+}
+
+func moveOrAppend(tokens []string, value string) []string {
+	for t := range tokens {
+		if tokens[t] == value {
+			tokens = append(tokens[:t], tokens[t+1:]...)
```
Comment (f41gh7) [inline lib/protoparser/opentelemetry/stream/streamparser.go:419]: It'd be great to have a test for this case.

 I suppose, this case must move value to the end of token array:
input:
[t1,t2,t2,value,t3,t4], value
ouput:
[t1,t2,t2,t3,t4,value]

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Comment by valyala** [comment]
@AndrewChubatiuk , please see the follow-up commit fb42380ef374c394053b4cc4f2b083ed4afc7d09 .

**Pull request merged**